magicblock-committor-service/src/intent_executor/intent_executor.rs

```
@@ -0,0 +1,247 @@
+                persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+            }
+            Err(Error::FailedCommitPreparationError(crate::transaction_preperator::error::Error::FailedToFitError)) => {
+                let update_status = CommitStatus::PartOfTooLargeBundleToProcess;
+                persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+            }
+            Err(Error::FailedCommitPreparationError(crate::transaction_preperator::error::Error::TaskBuilderError(err))) => {
+                match err {
+                    crate::tasks::task_builder::Error::CommitTasksBuildError(_) => {
+                        let update_status = CommitStatus::Failed;
+                        persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+                    }
+                    crate::tasks::task_builder::Error::FinalizedTasksBuildError(_) => {
+                        // TODO: commit signature to set this
+                    }
+                 }
+            },
+            Err(Error::FailedCommitPreparationError(crate::transaction_preperator::error::Error::DeliveryPreparationError(_))) => {
+                // Persisted internally
+            },
+            Err(Error::FailedToCommitError {err: _, signature}) => {
+                // Commit is a single TX, so if it fails, all of commited accounts marked FailedProcess
+                let status_signature = signature.map(|sig| CommitStatusSignatures {
+                    process_signature: sig,
+                    finalize_signature: None
+                });
+                let update_status = CommitStatus::FailedProcess(status_signature);
+                persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+            }
+            Err(Error::FailedFinalizePreparationError(_)) => {
+                // Not supported in persistor
```

OK, but we cannot silently ignore it?